### PR TITLE
Cas 31 staff idams login

### PR DESCRIPTION
**Pull request by witpo**

### CODE REVIEW

File: src/SFA.DAS.Tools.Support.Web/App_Start/AuthenticationExtension.cs [hunk 1]
```diff
@@ -14,27 +15,30 @@ public static class AuthenticationExtension
...
+                options.DefaultSignInScheme = CookieAuthenticationDefaults.AuthenticationScheme;
+                options.DefaultChallengeScheme = WsFederationDefaults.AuthenticationScheme;
+                options.DefaultSignOutScheme = WsFederationDefaults.AuthenticationScheme;
+            })
+             .AddWsFederation(options =>
+             {
+                 options.Wtrealm = authenticationConfiguration.Wtrealm;
+                 options.MetadataAddress = authenticationConfiguration.MetadataAddress;
+                 options.UseTokenLifetime = false;
+                 options.TokenValidationParameters.RoleClaimType = "TEST";
```
Comment (DannyGB) [inline src/SFA.DAS.Tools.Support.Web/App_Start/AuthenticationExtension.cs:33]: Seems odd, should it be "TEST"?

File: src/SFA.DAS.Tools.Support.Web/Startup.cs [hunk 2]
```diff
@@ -60,7 +60,7 @@ public void ConfigureServices(IServiceCollection services)
             {
                 var policy = new AuthorizationPolicyBuilder()
                     .RequireAuthenticatedUser()
-                    .RequireRole(_configuration["RequiredRole"])
+                    //.RequireRole(_configuration["RequiredRole"])
```
Comment (DannyGB) [inline src/SFA.DAS.Tools.Support.Web/Startup.cs:63]: I hope you can remember to add this back in when the staff IDAMS role has been added :smile:

**Comment by CodeWolf86** [comment]
Does this need to be removed / renamed? https://github.com/SkillsFundingAgency/das-tools-support/blob/CAS-31_staff_idams_login/src/SFA.DAS.Tools.Support.Web/Configuration/AzureAdConfiguration.cs

Think removed given there is a new Authentication Configuration.cs file?

File: src/SFA.DAS.Tools.Support.Web/Startup.cs (hunk 2 above)
Comment (witpo) [reply src/SFA.DAS.Tools.Support.Web/Startup.cs]: done

File: src/SFA.DAS.Tools.Support.Web/App_Start/AuthenticationExtension.cs (hunk 1 above)
Comment (witpo) [reply src/SFA.DAS.Tools.Support.Web/App_Start/AuthenticationExtension.cs]: removed

File: README.md [hunk 3]
```diff
@@ -22,9 +21,8 @@ To get the project running you will need to:
 ### Prerequisites
 
 * An IDE supporting .NetCore 3.1
-* The [Tools Service](https://github.com/SkillsFundingAgency/das-tools-service)
-* A valid GitHub account that is a member of the SkillsFundingAgency organisation
-* The "Support Portal" role added to your GitHub account
+* A valid Staff IDAMS account
+* The SCS role added to your IDAMS account
```
Comment (DannyGB) [inline README.md:25]: There a multiple roles, SCS and SCP. If you mention one you may as well mention both?

File: src/SFA.DAS.Tools.Support.Web/Views/Error/AccessDenied.cshtml [hunk 4]
```diff
@@ -11,4 +11,10 @@
...
     </div>
-</div>
\ No newline at end of file
+</div>
+
+@foreach (var item in @User.Claims.ToList())
+{
+    <div> @item.Type</div> 
+    <div> @item.Value </div>
+}
```
Comment (DannyGB) [inline src/SFA.DAS.Tools.Support.Web/Views/Error/AccessDenied.cshtml:20]: This will be removed prior to it going into production?

**Review by DannyGB — COMMENTED** [review]
Could this branch be merged directly to master, rather than going through Chris's AAD branch?

File: README.md (hunk 3 above)
Comment (witpo) [reply README.md:25]: SCP is not used at the moment, that is why I have not included it

File: src/SFA.DAS.Tools.Support.Web/Views/Error/AccessDenied.cshtml (hunk 4 above)
Comment (witpo) [reply src/SFA.DAS.Tools.Support.Web/Views/Error/AccessDenied.cshtml:20]: it will be deleted once production is working

Comment (witpo) [reply src/SFA.DAS.Tools.Support.Web/Views/Error/AccessDenied.cshtml]: removed

**Comment by DannyGB** [comment]
Closing as replaced by PR #18

**Pull request closed without merging**